pkg/cmd/controller/start.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"time"
+
+	clientset "github.com/xmudrii/etcdproxy-controller/pkg/client/clientset/versioned"
+	informers "github.com/xmudrii/etcdproxy-controller/pkg/client/informers/externalversions"
+	"github.com/xmudrii/etcdproxy-controller/pkg/controller/etcdproxy"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/xmudrii/etcdproxy-controller/pkg/cmd/controller/config"
+	"github.com/xmudrii/etcdproxy-controller/pkg/options"
+)
+
+func NewCommandEtcdProxyControllerStart(stopCh <-chan struct{}) *cobra.Command {
+	o := options.NewEtcdProxyOptions()
+
+	cmd := &cobra.Command{
+		Short: "Start EtcdProxyController",
+		Long:  "Start EtcdProxyController",
+		RunE: func(c *cobra.Command, args []string) error {
+			cfg, err := o.Config()
+			if err != nil {
+				return err
+			}
+
+			if err := RunController(cfg, stopCh); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+
+	o.AddFlags(cmd.Flags())
+	return cmd
+}
+
+// initControllerClientSets returns kubernetes clientset and etcdproxy clientset.
+func initControllerClientSets(kubeconfig *restclient.Config) (*kubernetes.Clientset, *clientset.Clientset, error) {
```

Comment:
nit: single use, single path helper functions aren't super useful.

Reply:
Updated functions for both Clientsets and Informers.